DOCS-2277 Browser Monitoring Edits (#1572)
DataDog/browser-sdk: packages/rum/README.md

## Overview

Datadog Real User Monitoring (RUM) enables you to visualize and analyze the real-time performance and user journeys of your application's individual users. To collect events, add the RUM Browser SDK to your browser application and configure what data is collected using initialization parameters.

## Setup

The RUM Browser SDK supports all modern desktop and mobile browsers including IE11. For more information, see the [Browser Support][8] table.

> To set up RUM Browser Monitoring, create a RUM application:

Author: That's fair, but we are encouraging users to set the environment and service names as part of the Instrument your application step, and if we are documenting one part, I feel like we should document it all.

1. In Datadog, navigate to the [**RUM Applications** page][1] and click the **New Application** button.
2. Enter a name for your application and click **Generate Client Token**. This generates a `clientToken` and an `applicationId` for your application.

> 3. Choose the instrumentation type for the RUM Browser SDK: [npm](#npm), or a hosted version ([CDN async](#cdn-async) or [CDN sync](#cdn-sync)).

Author: ```suggestion
3. Choose the installation type for the RUM Browser SDK: [npm](#npm), or a hosted version ([CDN async](#cdn-async) or [CDN sync](#cdn-sync)).
```

> 4. Define the environment name, service name, and deployed version for your application to use [Unified Service Tagging][19] for RUM & Session Replay. For more information, see [Tagging](#tagging).

Author: ```suggestion
4. Define the environment name, service name, and deployed version for your application to use unified service tagging for [RUM & Session Replay][19]. For more information, see [Tagging](#tagging).
```

5. Set the sampling rate of total user sessions collected and use the slider to set the percentage of total [Browser Premium][11] sessions collected. Browser Premium sessions include resources, long tasks, and replay recordings.

> 6. Click the **Session Replay Enabled** toggle to enable users with permissions to access [Session Replay][17].

Author: ```suggestion
6. Click the **Session Replay Enabled** toggle to enable [Session Replay][17] to record user sessions.
```

: Optional<br/>
**Type**: Number<br/>
**Default**: `100`<br/>

> The percentage of tracked sessions with [Browser Premium pricing][11] features: `100` for all, `0` for none. For more details about `premiumSampleRate`, see the [sampling configuration](#browser-rum-and-rum-premium-sampling-configuration).

Author: ```suggestion
The percentage of tracked sessions with [Browser Premium pricing][11] features: `100` for all, `0` for none. For more details about `premiumSampleRate`, see the [sampling configuration](#browser-and-rum-premium-sampling-configuration).
```

})
```

> ### Browser RUM and Browser Premium sampling configuration

Author: ```suggestion
### Browser and Browser Premium sampling configuration
```

This feature requires the RUM Browser SDK v3.6.0+.

When a session is created, RUM tracks it as either:

- [**Browser RUM**][11]: Only sessions, views, actions, and errors are collected. Calls to `startSessionReplayRecording()` are ignored.

> - [**Browser Premium**][11]: Everything from Browser RUM is collected, including resources, long tasks, and replay recordings. To collect replay recordings, call `startSessionReplayRecording()`.

Author: ```suggestion
- [**Browser Premium**][11]: Everything from Browser is collected, including resources, long tasks, and replay recordings. To collect replay recordings, call `startSessionReplayRecording()`.
```

> - [**Browser Premium**][11]: Everything from Browser RUM, including resources, long tasks, and replay recordings, is collected. To collect replay recordings, call `startSessionReplayRecording()`.

Reviewer: ```suggestion
- [**Browser Premium**][11]: Everything from Browser RUM is collected, including resources, long tasks, and replay recordings. To collect replay recordings, call `startSessionReplayRecording()`.
```

Two initialization parameters are available to control how the session is tracked:

- `sampleRate` controls the percentage of overall sessions being tracked. It defaults to `100%`, so every sessions is tracked by default.
- `premiumSampleRate` is applied **after** the overall sample rate, and controls the percentage of sessions tracked as Browser Premium. It defaults to `100%`, so every sessions is tracked as Browser Premium by default.

> To track 100% of your sessions as Browser RUM:

Author: ```suggestion
To track 100% of your sessions as Browser:
```

})
```

### Browser and Browser Premium sampling configuration

> This feature requires the RUM Browser SDK v3.6.0+.

Author: ```suggestion
This feature requires the RUM Browser SDK v3.0.0+. 

<div class="alert alert-info">
The RUM Browser SDK v4.10.2 introduces the `premiumSampleRate` initialization parameter, deprecating the `replaySampleRate` initialization parameter.
</div>
```

});
```

> The `premiumSampleRate` is a percentage of `sampleRate`. If you set `sampleRate` to 60 and `premiumSampleRate` to 50, 40% of sessions are dropped, 30% of sessions are collected as Browser and 30% of sessions are collected as Browser Premium. 

Reviewer 2: <!-- Praises highlight something positive. Try to leave at least one of these comments per review (if it exists :^)  -->
**👏 praise:** ‏thanks, that is clearer!

Author: Thanks @nicholas-devlin and @MirandaKapin for initiating the feedback!

});
```

> For example, to track 60% of overall user sessions and set 70% of those sessions as Browser Premium, set `sampleRate` to 60 and `premiumSampleRate` to 70. 

Author: ```suggestion
The `premiumSampleRate` is a percentage of `sampleRate`. If you set `sampleRate` to 60 and `premiumSampleRate` to 50, 40% of sessions are dropped, 30% of sessions are collected as Browser and 30% of sessions are collected as Browser Premium. 
```

```
datadogRum.init({
    ....
    sampleRate: 60,

>     premiumSampleRate: 70

Author: ```suggestion
    premiumSampleRate: 50
```

});
```

> Within the 60% of user sessions collected for your application, 18% are Browser and 42% are Browser Premium. 

Author: ```suggestion
```

});
```

> For example, to track only 50% of your overall sessions with half tracked as Browser RUM and the other half tracked as Browser Premium, set the `sampleRate` and the `premiumSampleRate` to 50:

Author: ```suggestion
For example, to track only 50% of your overall sessions with half tracked as Browser and the other half tracked as Browser Premium, set the `sampleRate` and the `premiumSampleRate` to 50. 25% of your sessions are Browser Premium. 
```

});
```

### Tagging

A service is an independent, deployable code repository that maps to a set of pages. 

- If your browser application was constructed as a monolith, your RUM application has one service name for the application. 
- If your browser application was constructed as separate repositories for multiple pages, edit the default service names throughout the lifecycle of your application. 

> For more information, see Configuring Browser Services (link here).

Author: Link TBD here @amina196

[8]: https://github.com/DataDog/browser-sdk/blob/main/packages/rum/BROWSER_SUPPORT.md
[9]: https://docs.datadoghq.com/real_user_monitoring/browser/tracking_user_actions#declaring-a-name-for-click-actions
[10]: https://docs.datadoghq.com/real_user_monitoring/browser/modifying_data_and_context/?tab=npm#override-default-rum-view-names
[11]: https://www.datadoghq.com/pricing/?product=real-user-monitoring--session-replay#real-user-monitoring--session-replay
[12]: https://docs.datadoghq.com/real_user_monitoring/connect_rum_and_traces?tab=browserrum
[13]: https://docs.datadoghq.com/real_user_monitoring/session_replay/privacy_options?tab=maskuserinput
[14]: https://docs.datadoghq.com/agent/basic_agent_usage#datadog-site
[15]: https://docs.datadoghq.com/getting_started/tagging/#defining-tags
[16]: https://docs.datadoghq.com/real_user_monitoring/browser/monitoring_page_performance/#how-page-activity-is-calculated
[17]: https://docs.datadoghq.com/real_user_monitoring/session_replay/
[18]: https://docs.datadoghq.com/real_user_monitoring/session_replay/privacy_options

> [19]: https://docs.datadoghq.com/getting_started/tagging/using_tags

Author: ```suggestion
[19]: https://docs.datadoghq.com/getting_started/tagging/using_tags/#rum--session-replay
```
